### ISSUE TITLE
Wrong scope for decorators variable

### ISSUE THREAD

**Issue by ascariandrea** [loganfsmyth/babel-plugin-transform-decorators-legacy#23]
I'm encountering a bug due to a wrong scope variable definition.
I have a component:

``` js
@props({
 label: React.PropTypes.string,
 className: React.PropTypes.string,
 style: React.PropTypes.object
})
class Component extends React.Component {
 render() {
  const { label, ...props } = this.props;

  return (
  <div {...props}>
   {label}
  </div> 
  );
 }
}
```

but in `render` method `props` variable is equal to `function es7PropsDecorator(type, options) { ... }`, while with `babel@^5` works just fine.

**Comment by loganfsmyth** [comment]
Can you show the transpiled output of that code? This sounds more like an unrelated Babel but, like maybe you are accessing the outer `props`(the decorator factory) instead of the inner `props` (the spread result. When I compile your snippet for myself it seems to work through.

What specifically are you seeing that makes you say `props variable is equal to function es7PropsDecorator`?

**Comment by ascariandrea** [comment]
I've created a dumb component:

``` js
import React from 'react';
import { props, t } from './src/utils';

@props({
  label: t.String,
  className: t.String,
  style: t.Object
})
class Label extends React.Component {
    render() {
      const {
        label,
        ...props
      } = this.props;

      console.log(props);

      return (
        <div {...props}>
          {label}
        </div>
      );
    }
}

const output = React.renderToString(
  <Label
    label={'This is a label!'}
    style={{ background: 'black', color: 'white' }}
    className='label'
  />
);
console.log(output);
```

and this is the full output:

``` js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _dec, _class;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _utils = require('./src/utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Label = (_dec = (0, _utils.props)({
  label: _utils.t.String,
  className: _utils.t.String,
  style: _utils.t.Object
}), _dec(_class = function (_React$Component) {
  _inherits(Label, _React$Component);

  function Label() {
    _classCallCheck(this, Label);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Label).apply(this, arguments));
  }

  _createClass(Label, [{
    key: 'render',
    value: function render() {
      var _props = this.props;
      var label = _props.label;

      var props = _objectWithoutProperties(_props, ['label']);

      console.log(_utils.props);

      return _react2.default.createElement(
        'div',
        _utils.props,
        label
      );
    }
  }]);

  return Label;
}(_react2.default.Component)) || _class);


var output = _react2.default.renderToString(_react2.default.createElement(Label, {
  label: 'This is a label!',
  style: { background: 'black', color: 'white' },
  className: 'label'
}));
console.log(output);
```

as you can see the `console.log(props)` is transpiled in `console.log(_utils.props)`.

**Comment by loganfsmyth** [comment]
Cool, this is not decorator related. I've filed it for you here: https://phabricator.babeljs.io/T7178

**Comment by ascariandrea** [comment]
Ok, sorry to bother you, but thanks!

**Comment by loganfsmyth** [comment]
@ascariandrea FYI the fix for this just went out in Babel 6.6.5

**Comment by ascariandrea** [comment]
Thanks @loganfsmyth!
